File: app/src/main/java/com/kgb/k_app/model/StringData.java
package com.kgb.k_app.model;

import android.content.ContentValues;

import com.kgb.k_app.data.BaseData;
import com.kgb.k_app.data.Data;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Krzysztof Betlej <kgbetlej@gmail.com>.
 * @date 2/22/17
 * @copyright Copyright (c) 2017 KGBetlej
 */

public class StringData extends BaseData {
    String mData;

    public StringData(String data) {
        mData = data;
    }


    @Override
    public long getId() {
        return 0;
    }

    @Override
    public ContentValues getData() {
        return null;
    }

    @Override
    public String getName() {
        return mData;
    }

    @Override
    public String getTableName() {
        return null;
    }

    public static List<StringData> asList(String[] source) {
        List<StringData> result = new ArrayList<>();
        for (String item : source) {
            result.add(new StringData(item));
        }
        return result;
    }
}

File: app/src/main/java/com/kgb/kapp/model/StringData.kt
package com.kgb.kapp.model

import android.content.ContentValues
import com.kgb.kapp.data.AbstractData
import java.util.ArrayList

/**
 * @author Krzysztof Betlej <kgbetlej></kgbetlej>@gmail.com>.
 * @date 2/22/17
 * @copyright Copyright (c) 2017 BitAge
 */

class StringData(override val name: String) : AbstractData() {

    override val id: Long
        get() = 0

    override fun  getTableName(): String {
        return ""
    }

    override fun getData(): ContentValues {
        val result = ContentValues()
        return result
    }

    companion object {
        @JvmStatic
        fun asList(source: Array<String>): List<StringData> {
            val result = ArrayList<StringData>()
            for (item in source) {
                result.add(StringData(item))
            }
            return result
        }
    }
}